### ISSUE TITLE
Stuck at Retrying setup: Initialization of YNCA receiver RX-A6A failed

### ISSUE THREAD

**Issue by mikesalz** [mvdwetering/yamaha_ynca#92]
**Describe the bug**
Trying to add my Yamaha receiver.  The integration finds it, but fails during configuration with an error of:
Retrying setup: Initialization of YNCA receiver RX-A6A failed

<img width="292" alt="image" src="https://user-images.githubusercontent.com/46855061/207386461-0da70209-ab0d-4007-81a6-c12c4a439344.png">

I believe the receiver is configured correctly to allow control:
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/46855061/207390592-e998f6be-0527-4716-9fc5-3fe383b17a34.png">

**Comment by mikesalz** [comment]
More info.  I enabled Debug Logging and received the following.  It looks like the problem has something to do with 2-channel audio.  I think this is the error:
> 2022-12-13 12:14:24.801 DEBUG (Thread-17) [ynca.connection] Connection closed/lost 'Auto' is not a valid TwoChDecoder

Full log:

> 
> 2022-12-13 12:14:20.554 DEBUG (Thread-17) [ynca.connection] Connected
> 2022-12-13 12:14:20.555 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:MODELNAME=?
> 2022-12-13 12:14:20.557 INFO (SyncWorker_1) [ynca.api] Subunit availability check start
> 2022-12-13 12:14:20.558 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:MODELNAME=RX-A6A
> 2022-12-13 12:14:20.655 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:MODELNAME=?
> 2022-12-13 12:14:20.657 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:MODELNAME=RX-A6A
> 2022-12-13 12:14:20.756 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:AVAIL=?
> 2022-12-13 12:14:20.758 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:20.857 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:AVAIL=?
> 2022-12-13 12:14:20.859 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:AVAIL=Ready
> 2022-12-13 12:14:20.958 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @ZONE2:AVAIL=?
> 2022-12-13 12:14:20.960 DEBUG (Thread-17) [ynca.connection] Recv - @ZONE2:AVAIL=Ready
> 2022-12-13 12:14:21.059 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @ZONE3:AVAIL=?
> 2022-12-13 12:14:21.061 DEBUG (Thread-17) [ynca.connection] Recv - @ZONE3:AVAIL=Ready
> 2022-12-13 12:14:21.160 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @ZONE4:AVAIL=?
> 2022-12-13 12:14:21.162 DEBUG (Thread-17) [ynca.connection] Recv - @ZONE4:AVAIL=Ready
> 2022-12-13 12:14:21.261 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @AIRPLAY:AVAIL=?
> 2022-12-13 12:14:21.263 DEBUG (Thread-17) [ynca.connection] Recv - @AIRPLAY:AVAIL=Not Ready
> 2022-12-13 12:14:21.362 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @BT:AVAIL=?
> 2022-12-13 12:14:21.364 DEBUG (Thread-17) [ynca.connection] Recv - @BT:AVAIL=Not Ready
> 2022-12-13 12:14:21.463 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @IPOD:AVAIL=?
> 2022-12-13 12:14:21.465 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:21.564 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @IPODUSB:AVAIL=?
> 2022-12-13 12:14:21.566 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:21.666 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @NAPSTER:AVAIL=?
> 2022-12-13 12:14:21.672 DEBUG (Thread-17) [ynca.connection] Recv - @NAPSTER:AVAIL=Not Ready
> 2022-12-13 12:14:21.766 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @NETRADIO:AVAIL=?
> 2022-12-13 12:14:21.769 DEBUG (Thread-17) [ynca.connection] Recv - @NETRADIO:AVAIL=Not Ready
> 2022-12-13 12:14:21.867 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @PANDORA:AVAIL=?
> 2022-12-13 12:14:21.869 DEBUG (Thread-17) [ynca.connection] Recv - @PANDORA:AVAIL=Not Ready
> 2022-12-13 12:14:21.968 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @PC:AVAIL=?
> 2022-12-13 12:14:21.970 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:22.069 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @RHAP:AVAIL=?
> 2022-12-13 12:14:22.071 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:22.171 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SIRIUS:AVAIL=?
> 2022-12-13 12:14:22.173 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:22.272 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SIRIUSIR:AVAIL=?
> 2022-12-13 12:14:22.274 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:22.373 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SIRIUSXM:AVAIL=?
> 2022-12-13 12:14:22.375 DEBUG (Thread-17) [ynca.connection] Recv - @SIRIUSXM:AVAIL=Not Ready
> 2022-12-13 12:14:22.474 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SERVER:AVAIL=?
> 2022-12-13 12:14:22.476 DEBUG (Thread-17) [ynca.connection] Recv - @SERVER:AVAIL=Not Ready
> 2022-12-13 12:14:22.576 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SPOTIFY:AVAIL=?
> 2022-12-13 12:14:22.578 DEBUG (Thread-17) [ynca.connection] Recv - @SPOTIFY:AVAIL=Not Ready
> 2022-12-13 12:14:22.677 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @TUN:AVAIL=?
> 2022-12-13 12:14:22.679 DEBUG (Thread-17) [ynca.connection] Recv - @TUN:AVAIL=Not Ready
> 2022-12-13 12:14:22.778 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @UAW:AVAIL=?
> 2022-12-13 12:14:22.779 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:22.879 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @USB:AVAIL=?
> 2022-12-13 12:14:22.881 DEBUG (Thread-17) [ynca.connection] Recv - @USB:AVAIL=Not Ready
> 2022-12-13 12:14:22.980 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:VERSION=?
> 2022-12-13 12:14:22.982 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:VERSION=1.80/3.12
> 2022-12-13 12:14:22.982 INFO (SyncWorker_1) [ynca.api] Subunit availability check done
> 2022-12-13 12:14:22.983 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.SYS initialization start.
> 2022-12-13 12:14:23.081 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:AVAIL=?
> 2022-12-13 12:14:23.083 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:23.182 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:INPNAME=?
> 2022-12-13 12:14:23.184 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:23.283 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:MODELNAME=?
> 2022-12-13 12:14:23.286 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:MODELNAME=RX-A6A
> 2022-12-13 12:14:23.384 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:PARTY=?
> 2022-12-13 12:14:23.386 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:PARTY=Off
> 2022-12-13 12:14:23.486 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:PARTYMUTE=?
> 2022-12-13 12:14:23.586 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:PWR=?
> 2022-12-13 12:14:23.588 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:PWR=On
> 2022-12-13 12:14:23.687 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:VERSION=?
> 2022-12-13 12:14:23.689 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:VERSION=1.80/3.12
> 2022-12-13 12:14:23.690 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.SYS initialization done.
> 2022-12-13 12:14:23.690 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.AIRPLAY initialization start.
> 2022-12-13 12:14:23.788 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @AIRPLAY:AVAIL=?
> 2022-12-13 12:14:23.791 DEBUG (Thread-17) [ynca.connection] Recv - @AIRPLAY:AVAIL=Not Ready
> 2022-12-13 12:14:23.889 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:VERSION=?
> 2022-12-13 12:14:23.891 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:VERSION=1.80/3.12
> 2022-12-13 12:14:23.892 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.AIRPLAY initialization done.
> 2022-12-13 12:14:23.892 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.BT initialization start.
> 2022-12-13 12:14:23.990 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @BT:AVAIL=?
> 2022-12-13 12:14:23.992 DEBUG (Thread-17) [ynca.connection] Recv - @BT:AVAIL=Not Ready
> 2022-12-13 12:14:24.091 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @SYS:VERSION=?
> 2022-12-13 12:14:24.093 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:VERSION=1.80/3.12
> 2022-12-13 12:14:24.094 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.BT initialization done.
> 2022-12-13 12:14:24.094 INFO (SyncWorker_1) [ynca.subunit] Subunit Subunit.MAIN initialization start.
> 2022-12-13 12:14:24.192 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:AVAIL=?
> 2022-12-13 12:14:24.194 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:AVAIL=Ready
> 2022-12-13 12:14:24.293 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:INITVOLLVL=?
> 2022-12-13 12:14:24.295 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:24.395 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:INITVOLMODE=?
> 2022-12-13 12:14:24.396 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:24.496 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:BASIC=?
> 2022-12-13 12:14:24.497 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:PWR=On
> 2022-12-13 12:14:24.499 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:SLEEP=Off
> 2022-12-13 12:14:24.500 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:VOL=-40.0
> 2022-12-13 12:14:24.502 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:MUTE=Off
> 2022-12-13 12:14:24.503 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:INP=HDMI2
> 2022-12-13 12:14:24.505 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:STRAIGHT=On
> 2022-12-13 12:14:24.506 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:ENHANCER=Off
> 2022-12-13 12:14:24.508 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:SOUNDPRG=Surround Decoder
> 2022-12-13 12:14:24.510 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:TONEBASS=0.0
> 2022-12-13 12:14:24.512 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:TONETREBLE=0.0
> 2022-12-13 12:14:24.514 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:PUREDIRMODE=Off
> 2022-12-13 12:14:24.516 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:HDMIOUT1=On
> 2022-12-13 12:14:24.517 DEBUG (Thread-17) [ynca.connection] Recv - @SYS:HDMIOUT2=On
> 2022-12-13 12:14:24.519 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:EXBASS=Off
> 2022-12-13 12:14:24.520 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:ADAPTIVEDRC=Off
> 2022-12-13 12:14:24.522 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:DIALOGUELVL=0
> 2022-12-13 12:14:24.524 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:DTSDIALOGUECONTROL=0
> 2022-12-13 12:14:24.597 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:MAXVOL=?
> 2022-12-13 12:14:24.599 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:24.698 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:SCENENAME=?
> 2022-12-13 12:14:24.700 DEBUG (Thread-17) [ynca.connection] Recv - @UNDEFINED
> 2022-12-13 12:14:24.799 DEBUG (Thread-18 (_send_handler)) [ynca.connection] Send - @MAIN:2CHDECODER=?
> 2022-12-13 12:14:24.801 DEBUG (Thread-17) [ynca.connection] Recv - @MAIN:2CHDECODER=Auto
> 2022-12-13 12:14:24.801 DEBUG (Thread-17) [ynca.connection] Connection closed/lost 'Auto' is not a valid TwoChDecoder

**Comment by mvdwetering** [comment]
From the info I found on the internet the 2CHDECODER only supported values below, so there was no "Auto" which the RX-A6A reports back  :/

```
    DolbyPl = "Dolby PL"
    DolbyPl2Movie = "Dolby PLII Movie"
    DolbyPl2Music = "Dolby PLII Music"
    DolbyPl2Game = "Dolby PLII Game"
    DolbyPl2xMovie = "Dolby PLIIx Movie"
    DolbyPl2xMusic = "Dolby PLIIx Music"
    DolbyPl2xGame = "Dolby PLIIx Game"
    DtsNeo6Cinema = "DTS NEO:6 Cinema"
    DtsNeo6Music = "DTS NEO:6 Music"
```

Based on a quick search through the manual it looks like the RX-A6A does not support any of the "Dolby PLII" or "DTS NEO" settings. I guess those are old by todays standards and replaced by something else. 

Anyway, I will update the reading of those values to be more forgiving so it at least continues initialization.

btw thanks for including the debug log. That was very helpful.

**Comment by mikesalz** [comment]
Hi @mvdwetering - Thanks for looking into this!  Please let me know if I can help test out any changes you make.

**Comment by mvdwetering** [comment]
I made a beta release which also should fix this issue. 
Could you try it and see if this was the only issue or if new ones start popping up?

In HACS you will have to enable beta releases by selecting the integration then in the 3-dot menu top right select "Redownload" and make sure "Show beta versions" is enabled. Then you can select it from the list (or when doing manual installs get it [here](https://github.com/mvdwetering/yamaha_ynca/releases/tag/v5.3.0b0))

**Comment by mikesalz** [comment]
It's looking better!  I was able to add the integration.  The power button works, and I can control Source and Sound Mode.  I did notice that Sound Mode includes some options that my receiver does not support, such as 7ch Stereo and 9ch Stereo.

Question: How would I set the scene?  Is that something that should show up in the UI or can it only be controlled through `yamaha_ynca.send_raw_ynca`?

Thanks for jumping on this so quickly!  It is VERY much appreciated!

<img width="999" alt="image" src="https://user-images.githubusercontent.com/46855061/208302843-23328a52-a925-4608-b740-3279e07f2d80.png">

<img width="560" alt="image" src="https://user-images.githubusercontent.com/46855061/208303049-1c311c99-7bac-48a4-aa89-a9cd6f44326a.png">

**Comment by mvdwetering** [comment]
It is unfortunately not possible to dynamically detect which soundmodes are supported by a receiver. You can hide the ones not supported by your receiver by pressing the "Configure" button on the integration card. 

That being said, I do keep a list of supported soundmodes per model so if you report back with the list of supported soundmodes I can add it to the ""database". Then the unsupported modes will get hidden automatically on a new install. The current list of known modes is [here](https://github.com/mvdwetering/ynca/blob/master/ynca/enums.py#L229)

---

For scenes I request the scene names and for all scenes that I get a name back I create a button entity to recall that scene. It looks like this for my receiver.
![image](https://user-images.githubusercontent.com/732514/208306966-cabae469-d8d2-4510-a3a1-ce7185c0d8b7.png)

If you don't have button entities then that detection is failing. Could you attach the diagostics file so I can see the initialization sequence?
![image](https://user-images.githubusercontent.com/732514/208306658-46e534a4-9915-4a29-bde7-b602474905b3.png)

Next to that could you check if recalling scenes is supported for your receiver? 
For example recalling scenes does not work for an RX-V475 while it reports the scene names and it does have the buttons on the front panel.

When you send the command `@MAIN:SCENE=Scene 1` with the `yamaha_ynca.send_raw_ynca` service  in the Developer tools it should activate scene one on the Main zone. Once confirmed working I can try to figure out how handle it further.

**Comment by mikesalz** [comment]
Below is a list of all of the Sound Modes for my RX-A6A.  I am guessing the other current Aventage models (A2A - A8A) would use the same modes.  I am also attaching the diagnostics.  I can't test the scenes at the moment because my 6 year old daughter is currently watching a Disney movie and she would beat me up if I started messing with it.  :)  I will check that out a little later today.

Hall in Munich
Hall in Vienna
Hall in Amsterdam
Church in Freiburg
Church in Royaumont
Chamber
Village Vanguard
Warehouse Loft
Cellar Club
The Roxy Theatre
The Bottom Line
Sports
Action Game
Roleplaying Game
Music Video
Recital Opera
Standard
Spectacle
Sci-Fi
Adventure
Drama
Mono Movie
Enhanced
2ch Stereo
All-Ch Stereo
Surround Decoder

[config_entry-yamaha_ynca-33911dd573770ab4f062bcd48efeb91a.json.txt](https://github.com/mvdwetering/yamaha_ynca/files/10254143/config_entry-yamaha_ynca-33911dd573770ab4f062bcd48efeb91a.json.txt)

**Comment by mikesalz** [comment]
Good news: Scenes are working using `yamaha_ynca.send_raw_ynca`!

**Comment by mvdwetering** [comment]
> ...list of soundmodes...

That list looks familiar except for the "Enhanced" one. Could you (while the integration is connected) select the  "Enhanced" soundmode on the receiver and then deselect it again. 

After that get the diagnostics file and report back. At the end of the diagnostics file there are the most recent commands sent/received so I can see how the commands for that one look and probably add support.

> Good news: Scenes are working using `yamaha_ynca.send_raw_ynca`!

Good news for you , but now I need to figure out a way on how to best handle "undetectable" scenes ;) 

Any ideas on how you would expect that to look in Home Assistant?

I am thinking about maybe putting some option in te Integration configuration to let users override the autodetect and specify the amount of scenes per zone.
Or I could just create disabled scene button entities if no scenes could be detected and let the user enable the ones that are supported.


FYI I found a bug with the integration while running my emujlator with the data from your diagnostics file. When you try to use Zone4 you will get errors because it does not support Volume control and I assumed that volume control would be always available. Fixing that now and will be in a later release.

**Comment by mikesalz** [comment]
I am hoping this diagnostics file gives you what you need.  It took me a minute to figure this out - I had Surround AI enabled, which I guess meant I could not manually select a different mode.  So I turned off Surround AI and then toggled between Enhanced and 2ch Stereo. 

As fas as undetectable scenes... I don't know.  Are they always named Scene 1 - Scene n?  Maybe provide a drop-down list of 10 scenes (just to pick a number) and let users remove the ones they don't need?  Personally, I am not as interested in a UI for it.  I just want to be able to programmatically change a scene based on which app is running on my Apple TV.  So if I open the Apple Music app, it can change to one scene and if I open Disney Plus it can switch to a different scene.

[config_entry-yamaha_ynca-33911dd573770ab4f062bcd48efeb91a.json.txt](https://github.com/mvdwetering/yamaha_ynca/files/10262933/config_entry-yamaha_ynca-33911dd573770ab4f062bcd48efeb91a.json.txt)

**Comment by The-Arsonist** [comment]
Jumping in on this to say thanks for your work on the integration, and I also have the scenes issue with my TSR-700, which is a Costco version of the RX-V6A. The source, volume, power, and sound mode integration in the main dashboard works as expected.

I have used the YNCA service in my automation to change scenes based on which Apple TV app is open, which is exactly what I was looking for. This thread was quite helpful in helping me (hopefully) get my receiver remote off my couch end table.

**Comment by mikesalz** [comment]
@The-Arsonist Have you already created automations based on your Apple TV?  I just got my Yamaha receiver and haven't created anything yet but I have some ideas.  Maybe you've thought of something I have not... Are you only changing scenes?  Are you doing anything else as well?

**Comment by The-Arsonist** [comment]
I'm just now diving into home assistant, so I haven't done much beyond adding devices yet. My main desire was to fix a lip-sync issue. All my streaming apps need 0 delay when running my Apple TV through the receiver, but my files in plex need around 350ms delay. I created an automation that is triggered by the entity state of the Apple TV by App ID to detect which app is being used. Then I use the Call Service action to have the YNCA service send the scene change command mentioned here. My scenes are set up for the same things, except the lip-sync value. It's currently working, I just need to figure out how to also have it unpause on the apple TV remote, because changing the scene pauses the playback for some reason.
My next automation will probably be to trigger lighting scenes based on apple TV playback, to turn off the lamps that cause reflections on my TV. Eventually I'll likely install some LED strip lighting that I will have slowly dim up when I pause playback.

**Comment by mvdwetering** [comment]
@mikesalz thanks for the diagnostics. It turns out the soundmode is also just called "Enhanced" in the protocol. I will add it to the list.

Scenes can/could have names like "BD/DVD Movie viewing" or "Radio listening" and the scene buttons were also labeled like that on the unit itself (like BD/DVD or Radio). From photos it looks like more recent models they are just numbered buttons. Which makes sense as having labels not match the function is confusing, so might as well make it more generic.
I think I will make it possible to override the autodetection of scenes an let users define the amount of scenes from the integration settings as that is where the other overrides for soundmodes and inputs also are now. It seems the most logical place.

**Comment by mikesalz** [comment]
@mvdwetering Yeah, on my remote the buttons are labeled with numbers.  The scene names in the receiver have default names like the ones you mentioned, but you can rename them to anything you'd like.

I noticed you bumped up the standard release.  Should I go back to that now or stick with the beta?

**Comment by mvdwetering** [comment]
The standard 5.3.0 release has the same code as the beta you are now on., so nothing functionally new.

But I would still recommend to just update to the latest version as I found out that I did not set the version in the beta relese (internally it has 0.0.0 instead of 5.3.0b0). Which I don't _think_ it is a problem, but lets avoid potential issues in the future :)

**Comment by mikesalz** [comment]
> I have used the YNCA service in my automation to change scenes based on which Apple TV app is open, which is exactly what I was looking for. 

@The-Arsonist Does your Apple TV pause any time you programmatically change a scene?  If so, have you found a way around that?  I think it is just the nature of the beast and there is probably no fix for it, but I figured I would ask to see if anyone has thought of something I have not.

@mvdwetering is there an attribute that shows which scene is currently selected?  Maybe I am just not seeing it because my receiver is not yet fully supported?

I hope everyone had a great xmas and New Year!!

**Comment by mvdwetering** [comment]
> @mvdwetering is there an attribute that shows which scene is currently selected? Maybe I am just not seeing it because my receiver is not yet fully supported?

There is no way to read back the active scene as far as I know. 
I think there is not even the concept of an active scene on the receivers. On recall it seems to just apply the settings stored in the scene and then it is done.

**Comment by peetersit** [comment]
Hi

@mvdwetering 
<img width="203" alt="image" src="https://user-images.githubusercontent.com/121575883/210273307-b0e72709-b1bc-4865-be0b-7755e661396b.png">

[home-assistant_yamaha_ynca_2023-01-02T19-55-41.700Z.log](https://github.com/mvdwetering/yamaha_ynca/files/10333307/home-assistant_yamaha_ynca_2023-01-02T19-55-41.700Z.log)


I'm fairly new to HomeAssistant (I'm coming from Homeseer) and currently trying to get HA to the same level as HS....I successfully added my RX-475 and as seen in the picture my RX-477 fails with the same error. How can i fixes this, I only need power on/off/standby.........so everything else can be stripped if needed..

**Comment by mvdwetering** [comment]
@peetersit could you please create a separate issue for your problem?
It because hard to keep track of things when they are all in one issue.

**Comment by peetersit** [comment]
@peetersit i made a new issue....

Thxz Patrick

**Comment by mvdwetering** [comment]
I released version 6.0.0 which has the ability to override the scenes autodetection from the integration configuration.
You can specify the amount scenes per zone.

**Comment by mikesalz** [comment]
Thanks for the update!  I will check it out!

**Comment by mikesalz** [comment]
Hi @mvdwetering - I noticed this in my logs.  It doesn't seem to be causing any real issues for me, but just wanted to point it out.

2023-01-18 17:06:54.923 WARNING (Thread-5) [ynca.enums] Unknown value 'AURO-3D' in TwoChDecoder

**Comment by mvdwetering** [comment]
The receiver is reporting a value for the Surround Decoder I have not seen before. The warning is there to detect mismatches as they might be a cause of issues. 
It seems newer receivers don't have the Dolby Prologic or DTS:Neo surround decoder modes anymore but new ones.

Could you cycle through the different modes and post back the Diagnostics file?
Then I can add those new values so they won't show as a warning anymore.

This seems to be the relevant section in the manual for Surround Decoder
https://manual.yamaha.com/av/20/rxa6a/en-US/314063371.html

Next to that one it would be nice if you could cycle through some of the other AURO-3D related options just to see if/what the receiver reports as I don't have those features on my own receiver.

AURO-3D modes listening mode
https://manual.yamaha.com/av/20/rxa6a/en-US/9496102667.html

and Auro-Matic preset
https://manual.yamaha.com/av/20/rxa6a/en-US/9496107275.html

**Comment by mvdwetering** [comment]
I added the "AURO-3D" value to the list of known values so the warning should not show up anymore. I could not add support for AURO-3D as I don't know all available modes, so if needed that could still be done at another time.

I think the initial issue has been solved, so I will close it. If something else comes up please open a new issue.
